catalog/spatial/registry/registry-federation-admin-impl/src/main/java/org/codice/ddf/registry/federationadmin/impl/AdminHelper.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.federationadmin.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.codice.ddf.registry.api.RegistryStore;
+import org.codice.ddf.ui.admin.api.ConfigurationAdminExt;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.metatype.ObjectClassDefinition;
+
+import ddf.catalog.service.ConfiguredService;
+import ddf.catalog.source.Source;
+
+public class AdminHelper {
+    private static final String REGISTRY_FILTER =
+            "(|(service.factoryPid=*Registry*Store*)(service.factoryPid=*registry*store*))";
+
+    private static final String MAP_ENTRY_ID = "id";
+
+    private static final String DISABLED = "_disabled";
+
+    private ConfigurationAdmin configurationAdmin;
+
+    public AdminHelper(ConfigurationAdmin configurationAdmin) {
+        this.configurationAdmin = configurationAdmin;
+    }
+
+    private BundleContext getBundleContext() {
+        Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+        if (bundle != null) {
+            return bundle.getBundleContext();
+        }
```

Comment:
In operation I don't think this can return null. Looks like a null return could happen only if this bundle's class loader didn't load the the current class which seems unlikely. Let me know if you think this still needs to be changed.